ci/jenkins_tests/entry_point.sh

```diff
@@ -0,0 +1,55 @@
+#!/usr/bin/env bash
+
+# Cause the script to exit if a single command fails.
+set -e
+
+# Show explicitly which commands are currently running.
+set -x
+
+RUN_DOC_TESTS=false
+RUN_TUNE_TESTS=false
+RUN_SGD_TESTS=false
+
+# Parse options
+while [[ $# -gt 0 ]]; do
+  key="$1"
+  case $key in
+    --run-doc)
+      RUN_DOC_TESTS=true
+      shift
+      ;;
+    --run-tune)
+      RUN_TUNE_TESTS=true
+      shift
+      ;;
+    --run-sgd)
+      RUN_SGD_TESTS=true
+      shift
+      ;;
+    *)
+      echo "ERROR: unknown option \"$key\""
+      exit -1
+      ;;
+  esac
+done
+
+MEMORY_SIZE="8G"
+SHM_SIZE="4G"
+
+ROOT_DIR=$(cd "$(dirname "${BASH_SOURCE:-$0}")"; pwd)
+
+DOCKER_SHA=$($ROOT_DIR/../../build-docker.sh --output-sha --no-cache)
```

Comment:
It's not strictly necessary for CI, but if it's not much trouble it'd be nice to quote absolute paths (like those containing `$ROOT_DIR`). If nothing else it'll help avoid copy-pasting it somewhere where it might cause a problem later. However, feel free to leave it as is too—I'll probably have go through and address all quoting issues in all scripts at some point later anyway.